Mathlib/AlgebraicGeometry/RationalMap.lean

```diff
@@ -230,6 +255,58 @@ lemma equiv_of_fromSpecStalkOfMem_eq [IrreducibleSpace X]
     simpa only [fromSpecStalkOfMem, restrict_domain, Opens.fromSpecStalkOfMem, Spec.map_inv,
       restrict_hom, Category.assoc, IsIso.eq_inv_comp, IsIso.hom_inv_id_assoc] using H
 
+instance (U : X.Opens) [IsReduced X] : IsReduced U := isReduced_of_isOpenImmersion U.ι
+
+lemma Opens.isDominant_ι {U : X.Opens} (hU : Dense (X := X) U) : IsDominant U.ι :=
+  ⟨by simpa [DenseRange] using hU⟩
+
+lemma Opens.isDominant_homOfLE {U V : X.Opens} (hU : Dense (X := X) U) (hU' : U ≤ V) :
+    IsDominant (X.homOfLE hU') :=
+  have : IsDominant (X.homOfLE hU' ≫ Opens.ι _) := by simpa using Opens.isDominant_ι hU
+  IsDominant.of_comp_of_isOpenImmersion (g := Opens.ι _) _
+
+/-- Two partial maps from reduced schemes to separated schemes are equivalent if and only if
+they are equal on **any** open dense subset. -/
+lemma equiv_iff_of_isSeparated_of_le [X.Over S] [Y.Over S] [IsReduced X]
+    [IsSeparated (Y ↘ S)] {f g : X.PartialMap Y} [f.IsOver S] [g.IsOver S]
+    {W : X.Opens} (hW : Dense (X := X) W) (hWl : W ≤ f.domain) (hWr : W ≤ g.domain) : f.equiv g ↔
+      (f.restrict W hW hWl).hom = (g.restrict W hW hWr).hom := by
+  refine ⟨fun ⟨V, hV, hVl, hVr, e⟩ ↦ ?_, fun e ↦ ⟨_, _, _, _, e⟩⟩
+  have : IsDominant (X.homOfLE (inf_le_left : W ⊓ V ≤ W)) :=
+    Opens.isDominant_homOfLE (hW.inter_of_isOpen_left hV W.2) _
+  apply ext_of_isDominant_of_isSeparated' S (X.homOfLE (inf_le_left : W ⊓ V ≤ W))
+  simpa using congr(X.homOfLE (inf_le_right : W ⊓ V ≤ V) ≫ $e)
+
+/-- Two partial maps from reduced schemes to separated schemes are equivalent if and only if
+they are equal on the intersection of the domains. -/
+lemma equiv_iff_of_isSeparated [X.Over S] [Y.Over S] [IsReduced X]
+    [IsSeparated (Y ↘ S)] {f g : X.PartialMap Y}
+    [f.IsOver S] [g.IsOver S] : f.equiv g ↔
+      (f.restrict _ (f.2.inter_of_isOpen_left g.2 f.domain.2) inf_le_left).hom =
+      (g.restrict _ (f.2.inter_of_isOpen_left g.2 f.domain.2) inf_le_right).hom :=
+  equiv_iff_of_isSeparated_of_le (S := S) _ _ _
+
+/-- Two partial map from reduced schemes to separated schemes with the same domain are equivalent
+if and only if they are equal. -/
+lemma equiv_iff_of_domain_eq_of_isSeparated [X.Over S] [Y.Over S] [IsReduced X]
+    [IsSeparated (Y ↘ S)] {f g : X.PartialMap Y} (hfg : f.domain = g.domain)
+    [f.IsOver S] [g.IsOver S] : f.equiv g ↔ f = g := by
+  rw [equiv_iff_of_isSeparated_of_le (S := S) f.dense_domain le_rfl hfg.le]
+  obtain ⟨Uf, _, f⟩ := f
+  obtain ⟨Ug, _, g⟩ := g
+  obtain rfl : Uf = Ug := hfg
+  simp
+
+/-- Two partial maps from reduced schemes to separated schemes are equivalent if and only if
+they are equal on the intersection of the domains. -/
```

Comment:
The docstring should be updated because `f` is a partial map but `g` is a morphism.